pypeit/images/combineimage.py

```diff
@@ -144,32 +196,43 @@ def run(self, bias=None, flatimages=None, ignore_saturation=False, sigma_clip=Tr
             print(msgs.indent() + strout.format("Filename", "Lamp status"))
             print(msgs.indent() + '-'*maxlen + "  " + '-'*maxlmp)
             for ff, file in enumerate(self.files):
-                print(msgs.indent() + strout.format(os.path.split(file)[1], " ".join(lampstat[ff].split("_"))))
+                print(msgs.indent()
+                      + strout.format(os.path.split(file)[1], " ".join(lampstat[ff].split("_"))))
             print(msgs.indent() + '-'*maxlen + "  " + '-'*maxlmp)
 
         # Coadd them
-        weights = np.ones(nimages)/float(nimages)
-        img_list = [img_stack]
         var_stack = utils.inverse(ivar_stack)
-        var_list = [var_stack, rn2img_stack]
         if combine_method == 'weightmean':
-            img_list_out, var_list_out, gpm, nused = combine.weighted_combine(
-                weights, img_list, var_list, (mask_stack == 0),
-                sigma_clip=sigma_clip, sigma_clip_stack=img_stack, sigrej=sigrej, maxiters=maxiters)
+            weights = np.ones(self.nfiles, dtype=float)/self.nfiles
+            img_list = [img_stack]
+            var_list = [var_stack, rn2img_stack]
+            img_list_out, var_list_out, gpm, nused \
+                    = combine.weighted_combine(weights, img_list, var_list, mask_stack==0,
+                                               sigma_clip=sigma_clip, sigma_clip_stack=img_stack,
+                                               sigrej=sigrej, maxiters=maxiters)
         elif combine_method == 'median':
-            # TODO This median combine should be using the mask!!!
-            img_list_out = [np.median(img_stack, axis=0)]
-            var_list_out = [np.median(var_stack, axis=0)]
-            var_list_out += [np.median(rn2img_stack, axis=0)]
-            gpm = np.ones_like(img_list_out[0], dtype='bool')
+            bpm_stack = mask_stack > 0
```

Comment:
So I don't think this is quite right, although better than what we had before since we are at least using the mask. The point is that if you median combine the images, then the noise on that combined image need to be computed. I don't see that error propagation being done here, although it is done correctly in combine.weighted_combine.  In practice we never median combine where the noise model matters, i.e. we just do this for flats and arcs where we don't use the noise, and where robust statistics are preferred. Anyway, you might think about how to report a better noise image here.

Reply:
Good catch!  I definitely didn't do this right.  Fixed.